File: leetCode/two-pointers/language/python/reverse-string-in-place.py
def reverseString(s):
    s = list(s)
    left = 0
    right = len(s) - 1
    while left < right:
        s[left], s[right] = s[right], s[left]
        left += 1
        right -= 1

    return "".join(s)


if __name__ == "__main__":
    print(reverseString("hello"))

File: leetCode/two-pointers/language/javascript/reverse-string-in-place.ts
function reverseString(str: string): string {
  const arr = str.split("");
  let i = 0;
  let j = arr.length - 1;
  while (i < j) {
    [arr[i], arr[j]] = [arr[j], arr[i]];
    i++;
    j--;
  }
  return arr.join("");
}

console.log(reverseString("hello"));
